src/overlay/widgets/TweakDBEditor.cpp

```diff
@@ -179,10 +182,39 @@ void TweakDBEditor::OnUpdate()
     }
 }
 
+void TweakDBEditor::RebuildCache()
+{
+    using namespace std::chrono_literals;
+
+    if (m_rebuildingCache)
+    {
+        assert(m_rebuildCacheFuture.valid());
+        if (m_rebuildCacheFuture.wait_for(3ms) != std::future_status::ready)
+            return;
+
+        m_rebuildCacheFuture.get();
+        m_rebuildingCache = false;
+        m_initialized = true;
+
+        return;
+    }
+
+    m_rebuildingCache = true;
+
+    m_rebuildCacheFuture = std::async(
```

Comment:
You need to set the policy to std::launch::async, otherwise you risk deferred being used as it's implementation defined

Reply:
good point, didnt think of that as async is used as default in vs

Reply:
added explicit launch type